src/ar/read_landmarks.cpp

```diff
@@ -76,14 +76,24 @@ void detectLandmarksLoop(){
 			}
 
 			landmark_lock.lock();
-			current_landmarks = output;
+			for (int i = 0; i < NUM_LANDMARKS; i++) {
+				if (output[i](2) != 0.0) current_landmarks[i] = output[i];
+			}
```

Comment:
I'm not familiar with the camera code so I definitely could be wrong about this, but wouldn't this cause stale data to stick around?

If the camera sees a target, but then the rover drives away, the next time somebody calls `readLandmarks()` it would return bad data, right?

This would be mitigated if clients call `readLandmark()` repeatedly, so there's not much time for data to become stale, but is that behavior we can we rely on?

Reply:
That was my question as well. I'm assuming we call `readLandmarks()` frequently enough that like a camera frame or two wouldn't make too much of a difference, but `current_landmarks` is updated by a loop in a separate thread as fast as the camera provides new frames, so theoretically, at about 30Hz. 

This seems like it's intended to mitigate the problem of the AR tag detection being spotty though - Nathan mentioned to me that tags would randomly sometimes be seen in one frame and not the next

Reply:
Ah, I just saw your comment on the same snippet. That makes sense, though, for why we have to do it. 

Instead of relying on the client to call `readLandmarks()` frequently, would it be cleaner to assign each data point a timestamp? That way, the landmark thread can detect expired data without relying on the client, and we would get the same benefits of working around the spotty detection.

Reply:
I agree that we shouldn't rely on the client to call `readLandmarks` to clear stale data. However, coming up with the right solution would involve a bit of discussion and is not a small change. How about for this PR, I'll add a comment explaining why we're doing things this way, and I'll make a Clickup task for coming up with a better solution.

I put it as a subtask of https://app.clickup.com/t/x7cd5v